python/lsst/consdb/pqserver.py

```diff
@@ -680,17 +701,32 @@ def insert_flexible_metadata(
         if dtype != type(value).__name__:
             raise BadValueException(f"{dtype} value", value)
 
+    has_multi_column_primary_keys = (
+        instrument_tables.get_schema_version(instrument_l) >= Version("3.2.0") and obs_type == "exposure"
+    )
+
     with engine.connect() as conn:
         for key, value in value_dict.items():
             value_str = str(value)
 
+            values = {"obs_id": obs_id, "key": key, "value": value_str}
+            if has_multi_column_primary_keys:
+                day_obs, seq_num = instrument_tables.get_day_obs_and_seq_num(instrument_l, obs_id)
```

Comment:
Doesn't seem good to call this for every key/value pair. Move it outside the loop?

Reply:
I don't think that works because there is no guarantee the day_obs and seq_num will be the same for all items.

Reply:
If `instrument_l` and `obs_id` are constant, `day_obs`/`seq_num` have to be constant.  The only things that are changing are `key` and `value`.